### PR TITLE
board/common/partitions : Refactoring

### PR DESCRIPTION
**Pull request by hasw7569**
Refactoring for improved readability and SAM CC entries

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [028f4bbc0e93d8e1926fe38482e3d7c04927e54a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [028f4bbc0e93d8e1926fe38482e3d7c04927e54a] - Code Rule Check (C++) OK.

File: os/board/common/partitions.c [hunk 1]
```diff
@@ -63,6 +63,90 @@ FAR struct mtd_dev_s *mtd_initialize(FAR struct mtd_geometry_s *geo)
 #endif
 	return mtd;
 }
+
+void type_specific_initialize(FAR struct mtd_dev_s *mtd_part, int partno, const char *types)
```
Comment (jeongchanKim) [inline os/board/common/partitions.c]: Please add static for internal functions.

File: os/board/common/partitions.c [hunk 2]
```diff
@@ -63,6 +63,90 @@ FAR struct mtd_dev_s *mtd_initialize(FAR struct mtd_geometry_s *geo)
...
+#endif
+#if defined(CONFIG_MTD_SMART) && defined(CONFIG_FS_SMARTFS)
+		if (!strncmp(types, "smartfs,", 8)) {
+			char partref[4];
+
+			snprintf(partref, sizeof(partref), "p%d", partno);
+			smart_initialize(CONFIG_FLASH_MINOR, mtd_part, partref);
+		} else
+#endif
+		{
```
Comment (jeongchanKim) [inline os/board/common/partitions.c]: Is this needed?

File: os/board/common/partitions.c (hunk 1 above)
Comment (hasw7569) [reply os/board/common/partitions.c]: Thank you.
I will add static for internal functions.

File: os/board/common/partitions.c (hunk 2 above)
Comment (hasw7569) [reply os/board/common/partitions.c]: I think it is not necessary.
I will modify this.

File: os/board/common/partitions.c [hunk 3]
```diff
@@ -63,6 +63,88 @@ FAR struct mtd_dev_s *mtd_initialize(FAR struct mtd_geometry_s *geo)
...
+			mtdconfig_register(mtd_part);
+		} else
+#endif
+#if defined(CONFIG_MTD_SMART) && defined(CONFIG_FS_SMARTFS)
+		if (!strncmp(types, "smartfs,", 8)) {
+			char partref[4];
+
+			snprintf(partref, sizeof(partref), "p%d", partno);
+			smart_initialize(CONFIG_FLASH_MINOR, mtd_part, partref);
+		} else
```
Comment (jeongchanKim) [inline os/board/common/partitions.c]: Please remove unnecessary `else`.

**Comment by seinfra** [comment]
Target : [7613ecb823e4ec66d82958ef9de00e2f3fd07191] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7613ecb823e4ec66d82958ef9de00e2f3fd07191] - Code Rule Check OK.

File: os/board/common/partitions.c (hunk 3 above)
Comment (hasw7569) [reply os/board/common/partitions.c]: modify

**Comment by seinfra** [comment]
Target : [0f489b8fbcd307e87e5e86c1c28908eb244d822d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0f489b8fbcd307e87e5e86c1c28908eb244d822d] - Code Rule Check (C++) OK.

File: os/board/common/partitions.c [hunk 4]
```diff
@@ -24,18 +24,18 @@
 #include <debug.h>
 #include <errno.h>
 #include <string.h>
-#include <sys/types.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <sys/types.h>
```
Comment (sunghan-chang) [inline os/board/common/partitions.c]: It is better to keep original or move up after <tinyara/config.h>. No need to move down.

File: os/board/common/partitions.c [hunk 5]
```diff
@@ -63,16 +63,98 @@ FAR struct mtd_dev_s *mtd_initialize(FAR struct mtd_geometry_s *geo)
...
+				break;
+			}
+		}
+		mtd_setpartitionname(mtd_part, part_name);
+	}
+}
+#endif
+
+#ifdef CONFIG_BINARY_MANAGER
+static void binary_manager_register(int partno, char *part_name, int partsize, const char *types)
```
Comment (sunghan-chang) [inline os/board/common/partitions.c]: let's rename like binary_manager_update_partition_info.

File: os/board/common/partitions.c [hunk 6]
```diff
@@ -110,77 +192,16 @@ void configure_partitions(void)
...
 #endif
-
-		/* Move to next part size information. */
-		while ((*parts++) != ',');
-
-		/* Move to next part type information. */
-		while ((*types++) != ',');
-
-		partno++;
+		move_to_next_part(&parts, &types, &partno);
```
Comment (YiHyunJin) [inline os/board/common/partitions.c]: `
static void move_to_next_part(const char **parts)
{
		/* Move to next  information. */
		while (*(*parts)++ != ',');
}
`
I think this is better.
After it is good to use in while (*(names++) != ',');, while ((*parts++) != ','); and while ((*types++) != ',');.
Like below
`
move_to_next_part(&name);
move_to_next_part(&parts);
move_to_next_part(&types);
`

Comment (hasw7569) [reply os/board/common/partitions.c]: I agree with your opinion.
I will fix it.

File: os/board/common/partitions.c (hunk 4 above)
Comment (hasw7569) [reply os/board/common/partitions.c]: modify

File: os/board/common/partitions.c (hunk 5 above)
Comment (hasw7569) [reply os/board/common/partitions.c]: modify

**Comment by seinfra** [comment]
Target : [a2ceaf75ca571a199cbb136d43c0a5bcc4351fea] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a2ceaf75ca571a199cbb136d43c0a5bcc4351fea] - Code Rule Check (C++) OK.

**Pull request merged**